week5/hyungyu/src/FilterableProductTable.jsx

```diff
@@ -0,0 +1,48 @@
+import React, { useState } from 'react'
+import SearchBar from './SearchBar'
+import ProductTable from './ProductTable'
+import InputBar from './inputBar'
+
+const FilterableProductTable = () => {
+  const [product, setProduct] = useState([
+    {id: "1", category: "Fruits", price: 1, stocked: true, name: "Apple"},
+    {id: "2", category: "Fruits", price: 1, stocked: true, name: "Dragonfruit"},
+    {id: "3", category: "Fruits", price: 2, stocked: false, name: "Passionfruit"},
+    {id: "4", category: "Vegetables", price: 2, stocked: true, name: "Spinach"},
+    {id: "5", category: "Vegetables", price: 4, stocked: false, name: "Pumpkin"},
+    {id: "6", category: "Vegetables", price: 1, stocked: true, name: "Peas"}
+  ]);
+
+  const [filterText, setFilterText] = useState("");
+  const [inStockOnly, setInStockOnly] = useState(false);
+
+  const addProduct = (newProduct) => {
+    setProduct((previousData) => [...previousData, newProduct]);
+  };
+
+  const deleteProduct = (deletedProductId) => {
+    const deletedProducts = [];
+    for(let i = 0; i < product.length; i++){
+      if(product[i].id !== deletedProductId){
+        deletedProducts.push(product[i]);
+      }
+    }
```

Comment:
for 문으로 반복을 돌려서 검증하는 것도 좋지만, Javascript에는 선언적으로 반복을 돌릴 수 있는 map, filter, reduce 등의 고차 함수가 있습니다!

filter의 경우 콜백함수로 전달된 판단 구문을 iterable에 일일이 적용한 후, true가 나온 것들만 모아서 배열로 return 해줍니다

```suggestion
    const deletedProducts = product.filter(
      (item) => item.id !== deletedProductId
    );
```

이런 식으로 더 가독성 좋고 편리하게 사용 가능해요